python/cudf/cudf/_lib/scalar.pyx

```diff
@@ -60,74 +65,129 @@ cdef class Scalar:
         dtype : dtype
             A NumPy dtype.
         """
-
         value = cudf.utils.dtypes.to_cudf_compatible_scalar(value, dtype=dtype)
-
-        valid = value is not None
+        valid = not is_null_host_scalar(value)
 
         if dtype is None:
-            if value is None:
+            if not valid:
                 raise TypeError(
                     "dtype required when constructing a null scalar"
                 )
+            if isinstance(value, (np.datetime64, np.timedelta64)):
+                if np.isnat(value) and np.datetime_data(
+                    value.dtype
+                )[0] == 'generic':
+                    raise TypeError("Need a dtype to build a NaT Scalar")
```

Comment:
Do we actually handle `NaT` correctly? Numpy / Pandas use `min(int64)` as a sentinel value for it and have special cased the operations, but we have `null` to fit that capacity.

If someone gives us a `NaT` scalar should we just treat it as `null`?

Reply:
We do just use null for this as far as the user representation of the data. I think [the catch is here](https://github.com/rapidsai/cudf/blob/branch-0.17/python/cudf/cudf/core/column/datetime.py#L245-L260), basically we need to keep `cudf.Scalar('NaT', dtype=self.dtype)` around as `DatetimeColumn.default_na_value`, because otherwise when we drop to pandas it will overflow during the arrow step.

This is as opposed to filling with any other value which would result in discarding the mask.

Reply:
Understood. We should really get rid of `default_na_value` and the `to_gpu_array` method that uses it. It's another situation of us implicitly handling nulls where we shouldn't be and should defer down to the downstream application to provide.